docs/web/diff.md

```diff
@@ -0,0 +1,902 @@
+```
+Analyze and store:
+
+```sh
+CodeChecker check -o ./test_report_dir_updated -b "g++ example.cpp -c"
+CodeChecker store -n "Test Run Updated" ./test_report_dir_updated
+```
+
+Looking at the web GUI, you can observe that the report emitted by `deadcode.DeadStores` has a false positive review status, shown by the grey no entry sign :no_entry_sign: in the Review Status column. Note that we cleared the review status filter, which won't list false positive reports by default. Below is the "Reports" view for "Test Run":
+
+![image](https://github.com/Szelethus/codechecker/assets/23276031/3117b9fd-bda0-495f-94a3-1d8843f1c8f5)
+
+Below is the "Reports" view for "Test Run Updated":
+
+![image](https://github.com/Szelethus/codechecker/assets/23276031/97c1bb34-0b2e-4244-ae3c-9cdb7a2ffc9d)
+
+As a result, this report is not outstanding in either of the runs:
+
+![image](https://github.com/Szelethus/codechecker/assets/23276031/9201f853-967b-4d88-a074-7289572c6061)
+
+### Using `CodeChecker cmd diff`
+* Show outstanding reports that appeared from first analysis to the next (via `--new`):
+  ```sh
+  CodeChecker cmd diff --basename "Test Run" --newname "Test Run Updated" --new
+  ```
+  Expected output:
+  ```
+  ----==== Checker Statistics ====----
+  ---------------------------------------------------
+  Checker name         | Severity | Number of reports
+  ---------------------------------------------------
+  core.NullDereference | HIGH     |                 1
+  ---------------------------------------------------
+  ----=================----
+  ```
+* Show outstanding reports that disappeared from first analysis to the next (via `--resolved`): 
+  ```sh
+  CodeChecker cmd diff --basename "Test Run" --newname "Test Run Updated" --resolved
+  ```
+  Expected output:
+  ```
+  ----==== Checker Statistics ====----
+  ----------------------------------------------
+  Checker name    | Severity | Number of reports
+  ----------------------------------------------
+  core.DivideZero | HIGH     |                 1
+  ----------------------------------------------
+  ----=================----
+  ```
+* Show outstanding reports that remained from first analysis to the next (via `--unresolved`): 
+  ```sh
+  CodeChecker cmd diff --basename "Test Run" --newname "Test Run Updated" --unresolved
+  ```
+  Expected output (no checkers are seen):
+  ```
+  ----======== Summary ========----
+  ---------------------------------------------
+  Number of processed analyzer result files | 0
+  Number of analyzer reports                | 0
+  ---------------------------------------------
+  ----=================----
+  ```
+
+## Diff on tags or timestamps
+
+You can filter the analysis results or diff results based on which reports were outstanding _before_ a given timestamp. In the below image, on the web GUI, we diff the reports outstanding before the 1st of January, 2023 in "Test Run" against reports  outstanding before the 1st of January, 2023 in "Test Run Updated":
```

Comment:
Refer to the introduction section "server based branch history analysis".

If a run is continously updated with new analysis results, it can store the past report states of all previous analysis executions. This can be useful to analyze how the reports were fixed, classified to false positives, or introduced over the development timeline.